apollo-federation/src/query_plan/fetch_dependency_graph.rs

```diff
@@ -3488,6 +3488,7 @@ pub(crate) fn compute_nodes_for_tree(
     initial_node_path: FetchDependencyGraphNodePath,
     initial_defer_context: DeferContext,
     initial_conditions: &OpGraphPathContext,
+    check_cancellation: &impl Fn() -> Result<(), SingleFederationError>,
```

Comment:
Using `impl Fn` here makes the whole method generic and monomorphized separately for every distinct callback that can be passed. This has some compilation time and code size cost.

Assuming that `check_cancellation` isn’t called in very hot loop, would it be better to use `&dyn Fn()`? (Here and in other places with a similar argument)